primer/view_components
# feat: updates autocomplete api to prevent accessibility violations

Closes https://github.com/github/accessibility/issues/596.

## Description
Updates the AutoComplete API for an easier developer experience and to prevent accessibility violations.

## Changes
1. Removes the need to pass in an `input` and `label` slot - these will now be programmatically added based on the arguments that a user passes in. A user can still add an `input` slot if desired, for use cases such as adding a custom class. The `label` slot has been removed to prevent misuse, but can be added back if we find the need for custom classes or another need.
2. Adds `is_label_visible` parameter - allows a user to visually hide a label, but keeps it for screen readers
3. Removes `aria-label` and `aria-labelledby` options - these should not be needed if the component is used as stated in the documentation.

## Validation Steps
1. Navigate to https://view-components-n0olq29t2-primer.vercel.app/view-components/components/beta/autocomplete
2. Open each example and test for keyboard functionality, inspect the markup and ensure it's accurate, test on a screen reader to ensure proper announcements
3. Open Storybook
4. Verify all stories and controls related to AutoComplete are functional

## Notes
- I am new to Ruby, Rails and contributing to Primer, so please let me know if there are better ways of doing what I have ❤️
